plutus-tx-plugin/src/PlutusTx/PIRTypes.hs

```diff
@@ -2,11 +2,13 @@ module PlutusTx.PIRTypes where
 
 import PlutusIR qualified as PIR
 import PlutusTx.Annotation
+import PlutusTx.Code
 
 type PIRKind = PIR.Kind Ann
 type PIRType uni = PIR.Type PIR.TyName uni Ann
 type PIRTerm uni fun = PIR.Term PIR.TyName PIR.Name uni fun Ann
 type PIRProgram uni fun = PIR.Program PIR.TyName PIR.Name uni fun ()
+type PIRProgramDebug uni fun = PIR.Program PIR.TyName PIR.Name uni fun SrcSpans
```

Comment:
So... why not just do this always? I mean, does it actually _cost_ us anything noticeable to put these in?

Reply:
This seems like the key design decision. A lot of this PR would be simpler without it!

Reply:
In production we need `PIRProgram`, not `PIRProgramDebug`.
This implementation converts the latter to the former basically already at the very end (in `runCompiler`), so I don't see how it can be made simpler.

Reply:
I don't see where the problem comes up. We already throw away annotations when serialising, this just means that the annotations we throw away are bigger. Or to put it another way, `runCompiler` is not "the very end", that's "where `CompiledCode` gets used", and I claim that already handles this.

Reply:
> We already throw away annotations when serialising

Does that mean we didn't need `CompiledCodeDebug` in the first place?

Reply:
Quite probably. I don't have it _all_ loaded into my head, but my current expectation is that it would all work fine if you just changed the annotation type unilaterally.

Reply:
Ok. I'll think about it. I think the main issue right now is that we are not getting enough `SrcSpans`. Suggestions are welcome.

Reply:
Poke around in GHC or ask in #ghc is probably the easiest way. "Is there a way to get GHC to add src ticks for every sub-expression" is basically what you're asking, I think.

Reply:
@michaelpj Throwing away annotations when serialising means `decode . encode /= id`. Is that a problem?